### DISCUSSION TITLE
Any way to fetch AppDataDirectory path?

### DISCUSSION THREAD

**Discussion by ppipada** [wailsapp/wails#3811] [Q&A] [answered]
Hi,

I am developing an application using wails and go backend + next frontend.
I have a basic requirement, where I want to store some user data (not necessarily settings, but some generated data from app frontend) into a application directory. Like an application specific userdata directory. 
In electron this is available via `app.getPath('userData')`

Is there a way in wails to get this directory so that the application can manage additional files/folders in that directory? 

Note that my usecase needs access to the dir path only in backend. Frontend would call a api on backend to send whatever data it wants to send and backend will store it. 

Pardon me if this is trivial, but I am very new to wails and could not find info on it in the docs.

**Comment by leaanthony** [comment, marked as answer]
Not currently but it's a great suggestion. This is something we will look at for v3 👍

**Reply by ppipada** [reply]
@leaanthony Thank you for your reply.

As a workaround, I was able to get a dir path via [XDG](https://github.com/adrg/xdg) package. Specifically,  
`configFilePath, err := xdg.ConfigFile(os.strings.ToLower(AppTitle))`.

`os.MkdirAll` may need to be called on the path in certain cases.
 
And, Thank you very much for your great work !!! 
It is a breath of fresh air to move from electron/ts/js things to a Go based desktop app system.

**Reply by ansxuman** [reply]
@leaanthony Can I take on this enhancement? If yes, could you please let me know where exactly you'd expect it to be implemented—would it be as a plugin or somewhere else?

**Reply by leaanthony** [reply]
Amazing that you're willing to do this so thank you! 

Are you thinking of adding this to v2 or v3? For an enhancement like this, we'd need to ensure that v3 has parity.

I was thinking that we could extend the runtime to add `Path(selector Path) string`. The `Path` type would be an enum. How does that sound?

**Reply by ansxuman** [reply]
I was actually thinking about adding this for v3. Adding Path in runtime makes total sense.

**Reply by ansxuman** [reply]
Created a PR, let me know if any changes are needed https://github.com/wailsapp/wails/pull/3823

**Reply by leaanthony** [reply]
@ppipada - https://github.com/wailsapp/wails/pull/3823
